https errors when downloading things from github with the windows launcher

These seem to happen since the switch to GraalVM 20 / JDK 11 (`2.0.0-RC6-27` and after):
```text
> cs get --cache tmp-cache https://github.com/coursier/coursier/releases/download/v2.0.2/coursier
Downloading https://github.com/coursier/coursier/releases/download/v2.0.2/coursier
Downloaded https://github.com/coursier/coursier/releases/download/v2.0.2/coursier
download error: Caught javax.net.ssl.SSLHandshakeException: Received fatal alert: handshake_failure (Received fatal alert: handshake_failure) while downloading https://github.com/coursier/coursier/releases/download/v2.0.2/coursier
```

That also makes `cs setup` or `cs java` fail, when they try to download the jabba index from GitHub, or native launchers (`cs`, `bloop`, `sbtn`) from there.

More detailed exception:
```text
> cs-2.0.2 get --cache tmp-cache https://github.com/coursier/coursier/releases/download/v2.0.2/coursier -v -v
Downloading https://github.com/coursier/coursier/releases/download/v2.0.2/coursier
Downloaded https://github.com/coursier/coursier/releases/download/v2.0.2/coursier
Exception in thread "main" coursier.cache.ArtifactError$DownloadError: download error: Caught javax.net.ssl.SSLHandshakeException: Received fatal alert: handshake_failure (Received fatal alert: handshake_failure) while downloading https://github.com/coursier/coursier/releases/download/v2.0.2/coursier
        at coursier.cache.FileCache$.helper$2(FileCache.scala:1017)
        at coursier.cache.FileCache$.coursier$cache$FileCache$$downloading(FileCache.scala:1031)
        at coursier.cache.FileCache.doDownload$1(FileCache.scala:321)
        at coursier.cache.FileCache.$anonfun$download$55(FileCache.scala:506)
        at coursier.cache.CacheLocks$.loop$1(CacheLocks.scala:72)
        at coursier.cache.CacheLocks$.withLockOr(CacheLocks.scala:98)
        at coursier.cache.FileCache.$anonfun$download$33(FileCache.scala:507)
        at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
        at scala.util.Success.$anonfun$map$1(Try.scala:255)
        at scala.util.Success.map(Try.scala:213)
        at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
        at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
        at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
        at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.lang.Thread.run(Thread.java:834)
        at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:517)
        at com.oracle.svm.core.windows.WindowsJavaThreads.osThreadStartRoutine(WindowsJavaThreads.java:138) 
Caused by: javax.net.ssl.SSLHandshakeException: Received fatal alert: handshake_failure
        at sun.security.ssl.Alert.createSSLException(Alert.java:131)
        at sun.security.ssl.Alert.createSSLException(Alert.java:117)
        at sun.security.ssl.TransportContext.fatal(TransportContext.java:313)
        at sun.security.ssl.Alert$AlertConsumer.consume(Alert.java:293)
        at sun.security.ssl.TransportContext.dispatch(TransportContext.java:186)
        at sun.security.ssl.SSLTransport.decode(SSLTransport.java:164)
        at sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1144)
        at sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1055)
        at sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:395)
        at sun.net.www.protocol.https.HttpsClient.afterConnect(HttpsClient.java:567)
        at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:185)
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1587)
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1515)
        at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:527)
        at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:334)
        at coursier.cache.CacheUrl$.redirectTo(CacheUrl.scala:153)
        at coursier.cache.CacheUrl$.redirect(CacheUrl.scala:163)
        at coursier.cache.CacheUrl$.urlConnectionMaybePartial(CacheUrl.scala:310)
        at coursier.cache.ConnectionBuilder.connectionMaybePartial(ConnectionBuilder.scala:45)
        at coursier.cache.FileCache.$anonfun$download$34(FileCache.scala:348)
        at coursier.cache.FileCache$.$anonfun$downloading$1(FileCache.scala:997)
        at coursier.cache.CacheLocks$.withUrlLock(CacheLocks.scala:116)
        at coursier.cache.FileCache$.helper$2(FileCache.scala:997)
        ... 18 more
```

--- a/.github/workflows/native-tests.yml
+++ b/.github/workflows/native-tests.yml
@@ -1,7 +1,8 @@
 name: Test native launchers
 on:
-  schedule:
-    - cron: "0 0 * * *"
+  push:
+    branches:
+    - master
 
 jobs:
 

--- a/build.sbt
+++ b/build.sbt
@@ -372,7 +372,8 @@ lazy val `cli-tests` = project("cli-tests")
           Def.task {
             val isWindows = System.getProperty("os.name").toLowerCase(java.util.Locale.ROOT).contains("windows")
             val accepts_J = !isWindows
-            (pack.in(cli, Compile).value.getAbsoluteFile./("bin/coursier").toString, Option("false"), Option(accepts_J.toString))
+            val packFile = if (isWindows) "bin/coursier.bat" else "bin/coursier"
+            (pack.in(cli, Compile).value.getAbsoluteFile./(packFile).toString, Option("false"), Option(accepts_J.toString))
           }
         case Some(launcher) =>
           Def.task((launcher, sys.props.get("coursier-test-launcher-accepts-D"), sys.props.get("coursier-test-launcher-accepts-J")))

--- a/modules/cli-tests/src/main/scala/coursier/clitests/GetTests.scala
+++ b/modules/cli-tests/src/main/scala/coursier/clitests/GetTests.scala
@@ -1,0 +1,67 @@
+package coursier.clitests
+
+import java.io.File
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Paths}
+
+import utest._
+
+abstract class GetTests extends TestSuite {
+
+  def launcher: String
+
+  val tests = Tests {
+    test("Maven Central") {
+      TestUtil.withTempDir { tmpDir =>
+        val cache = new File(tmpDir, "cache").getAbsolutePath
+        val output = LauncherTestUtil.output(
+          launcher,
+          "get",
+          "https://repo1.maven.org/maven2/io/get-coursier/coursier_2.13/2.0.2/coursier_2.13-2.0.2.pom.asc",
+          "--cache", cache
+        )
+        val content = new String(Files.readAllBytes(Paths.get(output.trim)), StandardCharsets.UTF_8)
+        val expectedContent =
+          """-----BEGIN PGP SIGNATURE-----
+            |
+            |iQEcBAABAgAGBQJfgK9TAAoJEM0bnY3ITspWmQgIAKlD5P5OsIfJs/O3d5ZtaaI6
+            |lrkdLZMS5i1eus7+qH7JMC8ZjaJ0/vM0AoXXLNFevjb51Xm1mmqMyyZVEKYm1RuE
+            |f7HQCOAmB0eqLVVTLRv3wYatMRxXYmBqkQpvUdRestYqiaK6eTTUPys3RDfgVjgr
+            |8vP1rkpWJ2c6IfVBmb1UhEMK4UFTryn0mVZ7dfBol8cb/+Z2rpksAygvzl1zTgvH
+            |0WR7ttpB93D+quQwo5iwqv0Bz3LPV/HD2CFY9C4lz0EsLFwYxvQtub3we4MEy6u+
+            |4HugYf3fFUZod+22Im0uh15ETvyqVqZFUXFPmTXN/P3QGcSIyJNitqI0DOrtTOw=
+            |=geJ4
+            |-----END PGP SIGNATURE-----
+            |""".stripMargin.replace("\r\n", "\n")
+        assert(content == expectedContent)
+      }
+    }
+    
+    test("GitHub") {
+      TestUtil.withTempDir { tmpDir =>
+        val cache = new File(tmpDir, "cache").getAbsolutePath
+        val output = LauncherTestUtil.output(
+          launcher,
+          "get",
+          "https://github.com/coursier/coursier/releases/download/v2.0.2/coursier.asc",
+          "--cache", cache
+        )
+        val content = new String(Files.readAllBytes(Paths.get(output.trim)), StandardCharsets.UTF_8)
+        val expectedContent =
+          """-----BEGIN PGP SIGNATURE-----
+            |
+            |iQEcBAABAgAGBQJfgLkhAAoJEM0bnY3ITspWwG4IAK4DNWn7nrGZGjf28e/r0Emi
+            |qMK0tA+jhtIEavEDpkQjmTD+hYKgYj2ixFPvNZrWqJ6JQAr2fJDAVQpe6HnP5eZe
+            |emMAx7wDZNE9VVVhDqqi3zQ25zcZJuD99eMJz690pE3eg7aflnAK9PhyJvpwZnx2
+            |TO8PuSqEy1dq2Ixa0jCUIWKX4deFEcjy60+kOtcNdBXHh2l/PTaDWtFi/EG3tfTv
+            |tHIic5ZkEDRBYBo8NOkBAvpuE4AiUj7/6/gJiwExAaA+0EnzRKERot7Y03ZB1Up7
+            |+czyB/AK6N1R7HGG7uVeUgG4O4TYqx501aU2zYjIFwMoclk+4NhFEAEcBby8DgE=
+            |=8g4q
+            |-----END PGP SIGNATURE-----
+            |""".stripMargin.replace("\r\n", "\n")
+        assert(content == expectedContent)
+      }
+    }
+  }
+
+}

--- a/modules/cli-tests/src/test/scala/coursier/clitests/PackGetTests.scala
+++ b/modules/cli-tests/src/test/scala/coursier/clitests/PackGetTests.scala
@@ -1,0 +1,5 @@
+package coursier.clitests
+
+object PackGetTests extends GetTests {
+  val launcher = LauncherTestUtil.launcher
+}

--- a/scripts/shared/GenerateLauncher.sc
+++ b/scripts/shared/GenerateLauncher.sc
@@ -294,15 +294,7 @@ def nativeImage(
   }
 
   val extraCsOpts = Seq("--java-opt", s"-Xmx$mem")
-  if (Util.os == "win") {
-    // getting weird TLS-related linking errors without this
-    val javaSecurityOverrides =
-      """security.provider.3=what.we.put.here.doesnt.matter.ButThisHasToBeOverridden
-        |""".stripMargin.getBytes
-    Util.withTmpFile("java.security.overrides-", ".properties", javaSecurityOverrides) { path =>
-      run(s"-J-Djava.security.properties=$path" +: extraNativeImageOpts, extraCsOpts)
-    }
-  } else if (Util.os == "linux" && coursierLauncher.endsWith("cs"))
+  if (Util.os == "win" || (Util.os == "linux" && coursierLauncher.endsWith("cs")))
     run(extraNativeImageOpts, extraCsOpts)
   else
     run(extraNativeImageOpts)

